add menu at the end of a game with a restart button

Joshix-1 opened the pull request:
# Description
I've added a better menu at the end of the games, that allows to restart or return to the menu.

![image](https://user-images.githubusercontent.com/57299889/177320963-5fe7628c-565d-49e4-a564-c44f1536b36e.png)

## Type of change
- [ ] Bug fix (non breaking change that fixes an issue)
- [x] New feature (non breaking change that adds functionality)
- [ ] Tooling (non breaking change that adds functionality to the workflow)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)

## Environment
Please describe your environment and remove nonapplicable environment information from this section.
- OS:  Arch Linux

## Testing
I played multiple games and looked at how the menu behaved.

# Checklist
- [x] My code follows the general style guidelines of the project
- [x] I have perfomed a self-review of my code
- [x] I have commented my code, particulary where it is unclear
- [x] My changes generate no new warnings or errors
- [x] The project compiles and runs correctly

ASecondGuy commented:
Your title has a typo.
I think calling end_game more than once should be possible.
If not every of the potentially hundreds of games needs to worry about that.
If the GameManager just ignored those calls this problem is solved for good.

Joshix-1 commented:
GameManager currently should ignore the calls

If it is called the second time it doesn't know the current game and returns early

ASecondGuy commented:
Then why is [this](https://github.com/letsgamedev/Suffragium/blob/15a8bd22aae867b1607e85596a31e12b6d8f3634/game/games/flappybird/player.gd#L25) needed?

Joshix-1 commented:
I don't know it. I just expierienced the bug, and that fixed it

Numenter commented on game/games/flappybird/player.gd:
```diff
@@ -47,6 +54,11 @@ func start_game():
 	started = true
 
 
+func end_game():
+	ended = true
+	GameManager.end_game(END_MESSAGE % score, score)
```
```suggestion
func end_game():
	set_physics_process(false)
	GameManager.end_game(END_MESSAGE % score, score)
```
Just disable physics_process?

Numenter commented:
> Then why is [this](https://github.com/letsgamedev/Suffragium/blob/15a8bd22aae867b1607e85596a31e12b6d8f3634/game/games/flappybird/player.gd#L25) needed?

> I don't know it. I just expierienced the bug, and that fixed it

Game:end_game -> GM:restart -> NewGame:Hallo -> OldGame:end_game -> OldGame:queue_free()
Something like that.

b7g commented:
There is interference between the pause menu and the end game menu.

https://user-images.githubusercontent.com/74007114/177458995-e8b46c26-8624-4b8a-9681-b59c1d205a4a.mp4

Joshix-1 commented:
fixed that. The end_game_menu doesn't display anymore after pressing restart in the pause menu

b7g commented:
When closing the window I still get this error:

```
E 0:00:26.416   add_child: Parent node is busy setting up children, add_node() failed. Consider using call_deferred("add_child", child) instead.
  <C++-Fehler>  Condition "data.blocked > 0" is true.
  <C++-Quellcode>scene/main/node.cpp:1135 @ add_child()
  <Stacktrace>  game_manager.gd:68 @ end_game()
                game_manager.gd:29 @ _notification()
```

and it halts with this message:

`Invalid set index 'text' (on base: 'Nil') with value of type 'String'.`

The trace:
```
end_game_menu.gd:34 - at function: show
game_manager.gd:69 - at function: end_game
game_manager.gd:29 - at function: _notification
```

Joshix-1 commented:
Should be fixed now too

Joshix-1 committed:
add menu add the end of a game with a restart button


just disable physics process instead of using a variable


don't show end_game_menu after restarting


fix error when closing window while in game

Joshix-1 commented:
Rebased all the changes and squashed them into one commit.
